fastscanner/services/indicators/service.py

```diff
@@ -136,18 +157,95 @@ async def handle(self, channel_id: str, data: dict[Any, Any]) -> None:
             if "timestamp" not in data:
                 logger.warning(f"Missing timestamp in message from {channel_id}")
                 return
+
             ts = pd.to_datetime(int(data["timestamp"]), unit="ms", utc=True).tz_convert(
                 LOCAL_TIMEZONE_STR
             )
             new_row = pd.Series(data, name=ts)
 
-            for indicator in self._indicators:
-                new_row = await indicator.extend_realtime(self._symbol, new_row)
-
-            self._handler.handle(self._symbol, new_row)
+            await self._add_to_buffer(new_row)
```

Comment:
Let's put a lock here to make add_to_buffer atomic. I think if we get two concurrent calls of add_to_buffer, weird things can happen.